[Reland][FSDP] Mixed precision enablement"

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #75236
* #75227
* __->__ #75024

Reland https://github.com/pytorch/pytorch/pull/74452

Issue was older nccl version does not support bf16. Will take an approach similar to https://github.com/pytorch/pytorch/pull/67843 to ensure test only runs with later nccl versions.

Original commit changeset: 99295ea4ff02

Original Phabricator Diff: D35000703

Differential Revision: [D35287501](https://our.internmc.facebook.com/intern/diff/D35287501/)

rohan-varma committed:
Back out "Revert D35000703: [WIP][FSDP] Mixed precision enablement"

Original commit changeset: 99295ea4ff02

Original Phabricator Diff: D35000703

Differential Revision: [D35287501](https://our.internmc.facebook.com/intern/diff/D35287501/)

[ghstack-poisoned]

mrshenli reviewed (approved):
> Issue was older nccl version does not support bf16. Will take an approach similar to https://github.com/pytorch/pytorch/pull/67843 to ensure test only runs with later nccl versions.

Do we need to mention this in our docs?

rohan-varma committed:
Update on "[Reland][FSDP] Mixed precision enablement""


Reland https://github.com/pytorch/pytorch/pull/74452

Issue was older nccl version does not support bf16. Will take an approach similar to https://github.com/pytorch/pytorch/pull/67843 to ensure test only runs with later nccl versions.

Original commit changeset: 99295ea4ff02

Original Phabricator Diff: D35000703

Differential Revision: [D35287501](https://our.internmc.facebook.com/intern/diff/D35287501/)

[ghstack-poisoned]

rohan-varma commented:
> > Issue was older nccl version does not support bf16. Will take an approach similar to #67843 to ensure test only runs with later nccl versions.
> 
> Do we need to mention this in our docs?

Yep, we should definitely add this. Filed https://github.com/pytorch/pytorch/issues/75215

rohan-varma committed:
Update on "[Reland][FSDP] Mixed precision enablement""


Reland https://github.com/pytorch/pytorch/pull/74452

Issue was older nccl version does not support bf16. Will take an approach similar to https://github.com/pytorch/pytorch/pull/67843 to ensure test only runs with later nccl versions.

Original commit changeset: 99295ea4ff02

Original Phabricator Diff: D35000703

Differential Revision: [D35287501](https://our.internmc.facebook.com/intern/diff/D35287501/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Reland][FSDP] Mixed precision enablement""


Reland https://github.com/pytorch/pytorch/pull/74452

Issue was older nccl version does not support bf16. Will take an approach similar to https://github.com/pytorch/pytorch/pull/67843 to ensure test only runs with later nccl versions.

Original commit changeset: 99295ea4ff02

Original Phabricator Diff: D35000703

Differential Revision: [D35287501](https://our.internmc.facebook.com/intern/diff/D35287501/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Reland][FSDP] Mixed precision enablement""


Reland https://github.com/pytorch/pytorch/pull/74452

Issue was older nccl version does not support bf16. Will take an approach similar to https://github.com/pytorch/pytorch/pull/67843 to ensure test only runs with later nccl versions.

Original commit changeset: 99295ea4ff02

Original Phabricator Diff: D35000703

Differential Revision: [D35287501](https://our.internmc.facebook.com/intern/diff/D35287501/)

[ghstack-poisoned]
